opentelemetry-exporter-geneva/geneva-uploader/src/ingestion_service/uploader.rs

```diff
@@ -0,0 +1,212 @@
+use chrono::{DateTime, Duration as ChronoDuration, Utc};
+
+use crate::config_service::client::{
+    GenevaConfigClient, GenevaConfigClientError, IngestionGatewayInfo,
+};
+use chrono::{Datelike, Timelike};
+use reqwest::{header, Client};
+use serde::Deserialize;
+use serde_json::Value;
+use std::collections::HashMap;
+use std::time::Duration;
+use thiserror::Error;
+use url::form_urlencoded::byte_serialize;
+use uuid::Uuid;
+
+/// Error types for the Geneva Uploader
+#[derive(Debug, Error)]
+pub(crate) enum GenevaUploaderError {
+    #[error("HTTP error: {0}")]
+    Http(#[from] reqwest::Error),
+    #[error("JSON error: {0}")]
+    Json(#[from] serde_json::Error),
+    #[error("Config service error: {0}")]
+    ConfigService(String),
+    #[allow(dead_code)]
+    #[error("Upload failed with status {status}: {message}")]
+    UploadFailed { status: u16, message: String },
+    #[allow(dead_code)]
+    #[error("Uploader error: {0}")]
+    Other(String),
+}
+
+impl From<GenevaConfigClientError> for GenevaUploaderError {
+    fn from(err: GenevaConfigClientError) -> Self {
+        // This preserves the original error message format from the code
+        GenevaUploaderError::ConfigService(format!("GCS error: {}", err))
```

Comment:
It's not always going to be a GCS error. Maybe just name is config or config client error?

Reply:
Updated. Thanks/